### ISSUE TITLE
prefer-template in template

### ISSUE THREAD

**Issue by StijnQCA** [vuejs/eslint-plugin-vue#865]
**Please describe what the rule should do:**

The rule should have the template comply to the following rule: https://eslint.org/docs/rules/prefer-template
Currently when using this rule template-strings are not enforced inside the template

**What category should the rule belong to?**

- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<template>
    <div class="options">
        <button v-for="option in options" 
                      :class="'btn btn-' + option.style"
        >
            {{ option.text }}
        </button>
    </div>
</template>

<template>
    <h3 :style="'color:' + preferredTitleColor + '; font-size:' + preferredFontSize + ';'">
        Welcome
    <h3/>
</template>
```

**Additional context**

Nope

**Comment by ota-meshi** [comment]
I have implemented this rule. #1141
I plan to alpha release this weekend.
